tests/unit_test/channels/whatsapp_test.py

```
@@ -690,4 +691,445 @@ async def test_payment_message_request(self):
+                            "details": "Failed to send message because this user's phone number is part of an experiment"
+                        },
+                        "href": "https://developers.facebook.com/docs/whatsapp/cloud-api/support/error-codes/",
+                    }
+                ]
+                assert log.get("recipient") == "15551234567"
+
+    @pytest.mark.asyncio
+    async def test_whatsapp_valid_unsupported_message_request(self):
+        from kairon.chat.handlers.channels.whatsapp import Whatsapp, WhatsappBot
+        with patch.object(WhatsappBot, "mark_as_read"):
+            with patch.object(Whatsapp, "process_message") as mock_message:
```

Ruff

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)